Issue 43 create fuctional test stub

vladislav-kulakovsky opened the pull request:
I don't understand should I add @Before and @After sections?

vladislav-kulakovsky committed:
Add MongooseContainer like in hdfs storage driver. Add ContainerOutputCallback.java from hdfs driver

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"

vladislav-kulakovsky committed:
MongooseContainer Added

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"

vladislav-kulakovsky committed:
Merge branch 'master' into issue-43-create-fuctional-test-stub

vladislav-kulakovsky committed:
LogAnalyzer, LogPatterns, EnvUtil, PortTools added

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"

vladislav-kulakovsky committed:
Add test_stub.js cause SCENARIO_FILE = null faild with exception, but I expect default scenario

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/pravega/util/docker/MongooseContainer.java:
```diff
@@ -0,0 +1,206 @@
+		}
+		final CreateContainerResponse container = dockerClient
+				.createContainerCmd(testingImageId)
+				.withName("mongoose")
```
should be `mongoose-storage-driver-pravega`
https://hub.docker.com/r/emcmongoose/mongoose-storage-driver-pravega/

vladislav-kulakovsky replied:
@akurilov I had done it using an [example](https://github.com/emc-mongoose/mongoose-storage-driver-hdfs/blob/e2d977c809022d05c1cde39324fa5d56f1997af0/src/test/java/com/emc/mongoose/storage/driver/hdfs/util/docker/MongooseContainer.java#L152) hdfs driver

Could you explain why in hdfs used name `mongoose` but in pravega I should be used `mongoose-storage-driver-pravega`?

akurilov replied:
I was wrong. We build the testing image right here in this code above, so the name doesn't matter.

akurilov commented on src/test/resources/scenario/test_stub.js:
```diff
@@ -0,0 +1,31 @@
+                    }
+                }
+            },
+            "output": {
```
please remove the `output` section here as far as we will need the output log files for the validation

vladislav-kulakovsky replied:
Complete

akurilov reviewed (commented):
> I don't understand should I add @before and @after sections?

Nope. The static before/after "class" methods are used for this purpose

vladislav-kulakovsky committed:
`output` section removed from test_stub.js

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"

vladislav-kulakovsky committed:
master branch merged

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

"Signed-off-by: Vladislav Kulakovsky kulakovsky.vladislav@gmail.com"
